codegen/module_system.go

```diff
@@ -1049,6 +1098,45 @@ func (g *EndpointGenerator) ComputeSpec(
 	return endpointSpecs, nil
 }
 
+// UpdateQPSLevels updates map from client-method name to qps level
+func UpdateQPSLevels(yamlFile string) {
+	var config map[string]interface{}
+	file, err := ioutil.ReadFile(yamlFile)
+	if err != nil {
+		fmt.Printf("error reading file to update qps level")
+	}
+	if err == nil {
+		err = yaml.Unmarshal(file, &config)
+	}
+	// checks if yaml has required fields to update qpsLevels map
+	// prevents qpsLevel being "0" if no qpsLevel field
+	clientMethod, methodOK := config["clientMethod"]
+	clientID, clientOK := config["clientId"]
+	qpsLevel, qpsOK := config["qpsLevel"]
+	if methodOK && clientOK && qpsOK && err == nil {
+		// edge case where clientID or clientMethod is nil
+		if clientID != nil && clientMethod != nil {
+			// unique key because of potential clients having same method names (staging)
+			key := clientID.(string) + "-" + clientMethod.(string)
+			currentQPSLevel := qpsLevels[key]
+			// store highest qps level for circuit breaker in qpsLevels map
+			thisQPSLevel := int(qpsLevel.(float64))
+			if thisQPSLevel > currentQPSLevel {
```

Comment:
This can be something like this:

```
mutex.Lock()
if ()
 {
}
else {
}
mutex.Unlock()
```

That way we are doing lock/unlock only once.